storage/PoolImpl.cc

```diff
@@ -118,11 +118,64 @@ namespace storage
     }
 
 
+    unsigned long long
+    Pool::Impl::max_partition_size(Devicegraph* devicegraph, unsigned int number) const
+    {
+	ST_CHECK_PTR(devicegraph);
+
+	if (number == 0)
+	    ST_THROW(Exception("illegal value of zero for number"));
+
+	// For each partitionable find the biggest unused region.
+
+	vector<unsigned long long> candidates;
+
+	for (Partitionable* partitionable : get_partitionable_candidates(devicegraph))
+	{
+	    PartitionTable* partition_table = partitionable->get_partition_table();
```

Comment:
Can this return NULL for a disk without a partition table (but maybe a filesystem directly on it)?
In that case, a check is missing to prevent a segfault.

If `get_partitionable_candidates()` does not return such disks, nevermind (but then a comment might be useful mentioning that).

Reply:
Added a note.